src/features/class-parsely-support.php

```diff
@@ -45,8 +45,8 @@ public function boot(): void {
 	 * @return array<number> Array of post IDs.
 	 */
 	public function add_parsely_trending_posts_query( array $posts, array $args ): array {
-		$parsely = $GLOBALS['parsely'];
-		if ( ! $parsely->api_secret_is_set() ) {
+		$parsely = $GLOBALS['parsely'] ?? null;
+		if ( is_null( $parsely ) || ! $parsely->api_secret_is_set() ) {
```

Comment:
could do an instanceof check to be extra sure and get the type checking included